rootfs/startup.sh

```diff
@@ -13,51 +13,78 @@ if [ -n "$X11VNC_ARGS" ]; then
 fi
 
 if [ -n "$OPENBOX_ARGS" ]; then
-    sed -i "s#^command=/usr/bin/openbox.*#& ${OPENBOX_ARGS}#" /etc/supervisor/conf.d/supervisord.conf
+    sed -i "s#^command=/usr/bin/openbox\$#& ${OPENBOX_ARGS}#" /etc/supervisor/conf.d/supervisord.conf
 fi
 
 if [ -n "$RESOLUTION" ]; then
     sed -i "s/1024x768/$RESOLUTION/" /usr/local/bin/xvfb.sh
 fi
 
-# USER=${USER:-root}
-# HOME=/root
-# if [ "$USER" != "root" ]; then
-#     echo "* enable custom user: $USER"
-#     useradd --create-home --shell /bin/bash --user-group --groups adm,sudo $USER
-#     if [ -z "$PASSWORD" ]; then
-#         echo "  set default password to \"ubuntu\""
-#         PASSWORD=ubuntu
-#     fi
-#     HOME=/home/$USER
-#     echo "$USER:$PASSWORD" | chpasswd
-#     cp -r /root/{.gtkrc-2.0,.asoundrc} ${HOME}
-#     [ -d "/dev/snd" ] && chgrp -R adm /dev/snd
-# fi
-PASSWORD=designsafeubuntu1234!!
-USER="ubuntu"
-echo "$USER:$PASSWORD" | chpasswd
-HOME=/home/$USER
-cp -r /root/{.gtkrc-2.0,.asoundrc} ${HOME}
-[ -d "/dev/snd" ] && chgrp -R adm /dev/snd
-chown -R $USER:$USER $HOME/.[^.]*
+USER=${USER:-root}
+HOME=/root
+USER_GROUP_ID=${USER_GROUP_ID:-1000}
+groupadd --gid $USER_GROUP_ID appgroup
+
+if [ "$USER" != "root" ]; then
+    echo "* enable custom user: $USER"
+    useradd --uid ${USER_ID:-1001} --create-home --shell /bin/bash --user-group --groups appgroup,adm,sudo $USER
+    usermod -g appgroup $USER
+    if [ -z "$PASSWORD" ]; then
+        echo "  set default password to \"ubuntu\""
+        PASSWORD=ubuntu
+    fi
+    HOME=/home/$USER
+    echo "$USER:$PASSWORD" | chpasswd
+    cp -r /root/{.config,.gtkrc-2.0,.asoundrc} ${HOME}
+
+    # chown all files and directories excluding mounted dirs
+    find $HOME \( -path $HOME/mydata -prune -o \
```

Comment:
I don't have a suggestion but wondering about this list (community/projects/published/public) and wondering what happens if we add something else in the future that we forget to add here?

Reply:
Totally. I was toying with some more dynamic solutions but decided to just roll with this hardcoded solution for now